Scripts/script-DBotPredictPhishingEvaluation.yml

```diff
@@ -0,0 +1,169 @@
+  demisto.log("Evaluate %d samples" % len(labeled_data))
+  res = demisto.executeCommand('DBotPredictTextLabel', {
+      'inputText': map(lambda x: x['txt'], labeled_data),
+      'modelListName': demisto.args()['modelListName']
+  })
+  if isError(res[0]):
+      return_error(res[0]['Contents'])
+
+
+  predictions = res[0]['Contents']
+
+  df = pd.DataFrame.from_dict(predictions)
+  df.loc[:, 'Original'] = pd.Series(map(lambda x: x['label'], labeled_data))
+
+  result = []
+  for i in xrange(60, 100, 5):
+      if i < 80:
+          if i % 10 == 0:
+              result.append(get_threshold_precision(df, float(i)/100, 'Label', 'Original', 'Probability'))
+      else:
+          result.append(get_threshold_precision(df, float(i)/100, 'Label', 'Original', 'Probability'))
+  table_headers = ['Threshold', 'Below Threshold', 'F1 Score', 'Precision', 'Recall']
+  hr1 = tableToMarkdown('Evaluation probablity with different thresholds', result, headers=table_headers)
+  result = []
+  for i in xrange(10, 50, 5):
+      result.append(get_threshold_precision(df, i, 'Label', 'Original', 'InputTextNumberOfTokens'))
+
+  hr2 = tableToMarkdown('Evaluation input text length with different thresholds', result, headers=table_headers)
+  precision_score = metrics.precision_score(df['Original'], df['Label'], average='micro')
+  recall_score = metrics.recall_score(df['Original'], df['Label'], average='micro')
+  f1_score = metrics.f1_score(df['Original'], df['Label'], average='micro')
+  eval_result = {
+      'Precision': precision_score,
+      'Recall': recall_score,
+      'F1': f1_score,
+      'Size': len(labeled_data)
+  }
+  demisto.results({
+      'ContentsFormat': formats['json'],
+      'Type': entryTypes['note'],
+      'Contents': eval_result,
+      'EntryContext': {
+          'DBotPredictPhishingEvaluation': eval_result
+      },
+      'HumanReadable': hr1 + "\n\n" + hr2,
+      'HumanReadableFormat': formats["markdown"]
+  })
+  details_string = 'F1:%.2f | Precision:%.2f | Recall:%.2f | SamplesCount: %d' % (f1_score, precision_score, recall_score, len(labeled_data))
+  demisto.executeCommand("setIncident", {'details': details_string})
+
+
+type: python
+tags:
+- ml
+- phishing
+comment: Evaluate phishing model created by text classification ml automation.
+enabled: true
+args:
+- name: incidentsQuery
+  description: Query of the phishing incidents.
+  defaultValue: created:>="3 days ago" and type:"Phishing"
+- name: maxNumberOfIncidents
+  description: Maximum number of incidents
+  defaultValue: "100"
+- name: emailTextKey
+  description: Incident key to extract email text
+  defaultValue: details
+- name: emailSubjectKey
+  description: Incident key to extract email subject
+  defaultValue: name
+- name: tagKey
+  description: Incident key to extract tag
+  defaultValue: closeReason
+- name: phishingLabels
+  description: 'Comma-separated values of email tags values and mapping. The script
```

Comment:
Make sure to update after @kirbles19

Reply:
We gonna re-document it and create also zendesk article for that